src/tests/integration/character-countdown.test.ts

```diff
@@ -0,0 +1,154 @@
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const progressCircle = characterCountdown.querySelector('circle.progress')! as SVGCircleElement
+      const remainingRatio = Number(progressCircle.style.strokeDashoffset) / Number(progressCircle.style.strokeDasharray)
+      const numCharacters = getPlainText(ensureActiveTab(mocks.getState()).feedbackState.editorState).length
+      const filledInRatio = numCharacters / 280
+      expect(remainingRatio + filledInRatio).to.closeTo(1, 0.00000000000001)
+    })
+
+    it('renders no text', () => {
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const text = characterCountdown.querySelector('svg > text')
+      expect(text).to.equal(null)
+    })
+
+    // TODO: Hank to fix bug in code
+    it('is a progressbar with the an aria-valuenow equal to the filledInRatio', () => {
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const numCharacters = getPlainText(ensureActiveTab(mocks.getState()).feedbackState.editorState).length
+      const filledInRatio = numCharacters / 280
+
+      expect(Number(characterCountdown.getAttribute('aria-valuenow'))).to.equal(Math.round(filledInRatio * 100))
+    })
+
+    it('does not disable posting', () => {
+      const postButton = mocks.app().querySelector('.post-btn')! as HTMLButtonElement
+      expect(postButton.disabled).to.equal(false)
+    })
+  })
+
+  describe('when approaching the character limit', () => {
+    const almost280Chars = '@zing.com ' + 'x'.repeat(260)
+    before(() => {
+      mocks.backgroundWindow.store.dispatchers.updateEditorState({
+        editorState: fromText(almost280Chars),
+      })
+    })
+
+    it('renders a more full progress circle', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const progressCircle = characterCountdown.querySelector('circle.progress')! as SVGCircleElement
+      const remainingRatio = Number(progressCircle.style.strokeDashoffset) / Number(progressCircle.style.strokeDasharray)
+      const numCharacters = getPlainText(ensureActiveTab(mocks.getState()).feedbackState.editorState).length
+      const filledInRatio = numCharacters / 280
+      expect(remainingRatio).to.closeTo(0, 0.1)
+      expect(filledInRatio).to.closeTo(1, 0.1)
+    })
+
+    it('has a .warning class', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      expect(characterCountdown.querySelectorAll('svg.warning')).to.have.length(1)
+    })
+
+    it('renders the number of characters remaining as a text element', () => {
+      // TODO: Hank to write this test
+      expect(mocks.app().querySelectorAll('.warning > text')).to.have.length(1)
```

Comment:
This should inspect the `.innerHTML` of the text element and assert its contents, which should be the number of characters remaining. 

```typescript
const text = mocks.app().querySelector('.character-countdown text')!
expect(text.innerHTML).to.equal('10')
```

Reply:
Thanks